corehq/apps/smsbillables/models.py

```diff
@@ -315,26 +324,37 @@ def create(cls, message_log, api_response=None):
         country_code, national_number = get_country_code_and_national_number(phone_number)
 
         if backend_instance is None or _sms_backend_is_global(backend_instance):
-            billable.gateway_fee = SmsGatewayFee.get_by_criteria(
-                backend_api_id,
-                direction,
-                backend_instance=backend_instance,
-                country_code=country_code,
-                national_number=national_number,
-            )
-            if billable.gateway_fee is not None:
-                conversion_rate = billable.gateway_fee.currency.rate_to_default
+            if backend_api_id == SQLTwilioBackend.get_api_id():
+                twilio_charges = cls._get_twilio_charges(
+                    billable, message_log, backend_instance
+                )
+                billable.direct_gateway_fee = twilio_charges.twilio_gateway_fee
+                billable.gateway_fee = twilio_charges.gateway_fee
+                currency = twilio_charges.currency
+            else:
+                billable.gateway_fee = SmsGatewayFee.get_by_criteria(
+                    backend_api_id,
+                    direction,
+                    backend_instance=backend_instance,
+                    country_code=country_code,
+                    national_number=national_number,
+                )
+                if billable.gateway_fee is not None:
+                    currency = billable.gateway_fee.currency
+                else:
+                    currency = None
+                    smsbillables_logging.error(
```

Comment:
we should make sure that this logger sends out emails on errors. now that we've shut down sentry i don't think we get emails from these anymore